src/util/handleWebhookEvent.ts

```diff
@@ -100,15 +175,25 @@ function isDocument(doc: SanityDocument | null | undefined): doc is SanityDocume
   return Boolean(doc && doc._id)
 }
 
-function validatePayload(payload: SanityWebhookBody | undefined): payload is SanityWebhookBody {
-  if (!payload || typeof payload.ids !== 'object') {
+export function validateWebhookPayload(
+  payload: SanityWebhookBody | undefined,
+): 'v1' | 'v2' | false {
+  if (!payload) {
     return false
   }
 
-  const {created, deleted, updated} = payload.ids
-  if (!Array.isArray(created) || !Array.isArray(deleted) || !Array.isArray(updated)) {
-    return false
+  // Let's test V2 first as those documents could also include an `ids` object
+  if ('__webhooksVersion' in payload && payload.__webhooksVersion === 'v2') {
+    return 'v2'
   }
 
-  return true
+  if ('ids' in payload && typeof payload.ids === 'object') {
```

Comment:
Theoretically `ids` could be null and return `object` here, so maybe check if it's truthy first to prevent that one edge-case from breaking when destructuring?

```suggestion
  if ('ids' in payload && payload.ids && typeof payload.ids === 'object') {
```